src/internal/oscal/generate.go

```diff
@@ -20,739 +17,339 @@ type BaseFlags struct {
-		models := make([]string, 0)
-		for _, item := range schema.OneOf {
-			models = append(models, item.TypeObject.Required...)
-		}
-		return models, nil
-	}
-
-	if len(schema.Required) > 0 {
-		return schema.Required, nil
+	// Format the bytes
+	typeBytes, err = format.Source(typeBytes)
+	if err != nil {
+		return typeBytes, err
 	}
 
-	return []string{}, fmt.Errorf("Top-level 'required' field not found or is not populated. Please verify the OSCAL JSON schema file is valid.")
+	return typeBytes, nil
 }
 
-// TODO: Look into full definition naming when processing multiple schemas - unsupported at this time
-// create a map with $id primary keys and the object values
-func generateUniqueIdMap(schema jsonschema.Schema) (map[string]jsonschema.SchemaOrBool, error) {
-	result := make(map[string]jsonschema.SchemaOrBool)
-
-	for definition, item := range schema.Definitions {
-		if strings.Contains(definition, "Datatype") {
-			result["#/definitions/"+definition] = item
-		} else if definition == "json-schema-definition" {
-			// Skipping - TODO: determine relevance in the golang structs
-			continue
-		} else {
-			result[*item.TypeObject.ID] = item
-		}
-
+// initBuild initializes the build process by adding the root schema to the ref queue and building the definitions map.
+func (c *GeneratorConfig) initBuild(schema *jsonschema.Schema) (err error) {
+	if schema.ID == nil {
+		return fmt.Errorf("unable to find $id in schema")
 	}
 
-	return result, nil
-}
-
-// getRequiredFields collects the required fields in each OSCAL model definition.
-func getRequiredFields(obj map[string]jsonschema.SchemaOrBool, structId string) map[string]bool {
-	requiredFields := make(map[string]bool)
-
-	if required := obj[structId].TypeObject.Required; required != nil {
-		for _, v := range required {
-			requiredFields[v] = true
-		}
-	}
-
-	return requiredFields
-}
-
-// formatStructTags formats Go struct tags.
-func formatStructTags(obj map[string]jsonschema.SchemaOrBool, structId string, key string, tags []string) []string {
-	requiredFields := getRequiredFields(obj, structId)
-
-	tagList := []string{}
-	for _, tag := range tags {
-		// If this field is not required, then add omitempty to tag.
-		if _, ok := requiredFields[key]; !ok {
-			tagList = append(tagList, fmt.Sprintf("%s:\"%s,%s\"", tag, key, "omitempty"))
-		} else {
-			tagList = append(tagList, fmt.Sprintf("%s:\"%s\"", tag, key))
+	if schema.Definitions != nil {
+		c.definitions, err = getDefinitionMap(*schema)
+		if err != nil {
+			return err
 		}
 	}
 
-	return tagList
-}
-
-// buildStructData loops through "properties" fields
-// and constructs data for Go structs.
-func buildStructData(prop map[string]jsonschema.SchemaOrBool, obj map[string]jsonschema.SchemaOrBool, structId string, tags []string, structData []string, modelMap map[string][]string) ([]string, error) {
-	keys := make([]string, 0, len(prop))
-
-	for k := range prop {
-		keys = append(keys, k)
+	if schema.Ref != nil {
```

Comment:
If we were to transition to only allowing complete schemas - would this ever evaluate to true? Not a change now but I want to track it either way.